src/connection.js

```diff
@@ -367,18 +367,25 @@ const SimulatedTransactionResponseValidator = jsonRpcResultAndContext(
   }),
 );
 
+type InnerInstruction = {
+  index: number,
+  instructions: (ParsedInstruction | PartiallyDecodedInstruction)[]
+};
+
 /**
  * Metadata for a confirmed transaction on the ledger
  *
  * @typedef {Object} ConfirmedTransactionMeta
  * @property {number} fee The fee charged for processing the transaction
+ * @property {Array<InnerInstruction>} An array of cross program invoked instructions
  * @property {Array<number>} preBalances The balances of the transaction accounts before processing
  * @property {Array<number>} postBalances The balances of the transaction accounts after processing
  * @property {Array<string>} logMessages An array of program log messages emitted during a transaction
  * @property {object|null} err The error result of transaction processing
  */
 type ConfirmedTransactionMeta = {
```

Comment:
I think we'll need both a `ParsedConfirmedTransactionMeta` and a `ConfirmedTransactionMeta`. `ParsedConfirmedTransactionMeta` can be used for `ParsedConfirmedTransaction` and `ConfirmedTransactionMeta` can be used for `ConfirmedTransaction`

And then we can have a `ParsedInnerInstruction` for parsed instructions and `InnerInstruction` for hex encoded instruction data